flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetWriters.java

```diff
@@ -450,7 +450,10 @@ public boolean hasNext() {
 
     @Override
     protected Object get(RowData struct, int index) {
-      return fieldGetter[index].getFieldOrNull(struct);
+      if (struct.isNullAt(index)) {
+        return null;
+      }
+      return this.fieldGetter[index].getFieldOrNull(struct);
```

Comment:
Let's keep the work to just Flink 1.14 for now, so comments don't get duplicated on many things.